webxr-api/device.rs

```diff
@@ -62,9 +63,8 @@ pub trait Device: 'static {
     /// Sets the event handling callback
     fn set_event_callback(&mut self, callback: Box<dyn EventCallback>);
 
-    /// Whether the device is still connected
-    fn connected(&mut self) -> bool;
-
     /// Quit the session
     fn quit(&mut self);
+
+    fn set_quitter(&mut self, quitter: Quitter);
```

Comment:
Hmm, this is annoying, it would be nice if we could just generate an `end` event for this, but that goes to script rather than the session thread. We could just use a polling+waker API, which is what Rust futures use.

We can just land this for now, perhaps with an issue to remind us to revisit it?

Reply:
The quitter is basically a glorified waker, scoped so that all it can do is send quit events

Reply:
I think we're going to need a general MainThreadWaker anyway, perhaps Quitter could just use that?